src/XLParser/ExcelFormulaGrammar.cs

```diff
@@ -251,32 +259,41 @@ public ExcelFormulaGrammar() : base(false)
             #region References
 
             Reference.Rule = ReferenceItem
-                | Reference + colon + Reference
-                | Reference + intersectop + Reference
-                | OpenParen + Union + CloseParen
+                | ReferenceFunctionCall
                 | OpenParen + Reference + PreferShiftHere() + CloseParen
                 | Prefix + ReferenceItem
-                | Prefix + UDFToken + Arguments + CloseParen
                 | DynamicDataExchange
                 ;
 
+            ReferenceFunctionCall.Rule =
+                  Reference + colon + Reference
+                | Reference + intersectop + Reference
+                | OpenParen + Union + CloseParen
+                | RefFunctionName + Arguments + CloseParen
+                //| ConditionalRefFunctionName + Arguments + CloseParen
+                ;
+
+            RefFunctionName.Rule = ExcelRefFunctionToken | ExcelConditionalRefFunctionToken;
+
             Union.Rule = MakePlusRule(Union, comma, Reference);
 
             ReferenceItem.Rule =
                 Cell
                 | NamedRange
-                | ReferenceFunction
                 | VRange
                 | HRange
                 | RefError
+                | UDFunctionCall
```

Comment:
I don't think that is going to work, because UDF's are allowed to have a prefix, while reference functions aren't I believe. It was something I needed to check with you still though.

    Sheet!MyFunction()  <-- valid
    Sheet!IF(TRUE,A1,A2) <--- invalid
    Sheet!INDEX(...) <--- I think this is invalid but I'm not sure

So originally I had both as ref functions and UDF's as ReferenceFunctionCall in referenceitem, but changed it to the curent things when I found out above.